2-projects/1-todo-list/script.js

```diff
@@ -17,9 +41,58 @@ function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
   // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  let input = document.querySelector(".form-control");
+  let inputValue = [
+    {
+      task: input.value,
+      completed: false,
+    },
+  ];
+
+  populateTodoList(inputValue);
+  deleteAllCompletedTodos();
+  input.value = "";
 }
 
 // Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
 function deleteAllCompletedTodos() {
   // Write your code here...
+  let checks = document.querySelectorAll(".fa-check");
+  let deleteBtns = document.querySelectorAll(".fa-trash");
+
+  checks.forEach((check) => {
+    check.addEventListener("click", (e) => {
```

Comment:
nice job getting onto the advanced challenge!
Rather than adding your button event listeners in this function try and keep your functions focused on the specific functionality they're trying to achieve (it would make more sense to add these event listeners when you are creating your todo list in your `populateTodoList` function). 
This function should just be focusing on deleting all completed todos, so all elements should already be created & have their event listeners set up by this point.
One way of doing this could be to check if your list item has the line through class by doing something like this:
`element.classList.contains(class)`  - if it has the line through class, you can remove the item from the list :)